src/main/java/edu/cornell/kfs/module/purap/batch/service/impl/JaggaerGenerateSupplierXmlServiceImpl.java

```diff
@@ -0,0 +1,161 @@
+import edu.cornell.kfs.module.purap.jaggaer.supplier.xml.Authentication;
+import edu.cornell.kfs.module.purap.jaggaer.supplier.xml.Header;
+import edu.cornell.kfs.module.purap.jaggaer.supplier.xml.JaggaerBuilder;
+import edu.cornell.kfs.module.purap.jaggaer.supplier.xml.Supplier;
+import edu.cornell.kfs.module.purap.jaggaer.supplier.xml.SupplierRequestMessage;
+import edu.cornell.kfs.module.purap.jaggaer.supplier.xml.SupplierSyncMessage;
+import edu.cornell.kfs.sys.CUKFSConstants;
+import edu.cornell.kfs.sys.service.CUMarshalService;
+import jakarta.xml.bind.JAXBException;
+
+public class JaggaerGenerateSupplierXmlServiceImpl implements JaggaerGenerateSupplierXmlService {
+    private static final Logger LOG = LogManager.getLogger();
+    private static final DateTimeFormatter DATE_FORMATTER = DateTimeFormat
+            .forPattern(CUKFSConstants.DATE_FORMAT_yyyy_MM_dd_T_HH_mm_ss_SSS_Z).withLocale(Locale.US);
+
+    private static final String JAGGAER_UPLOAD_FILE_NAME = "jaggaerSupplierUploadFile_";
+
+    private String jaggaerXmlDirectory;
+
+    protected DateTimeService dateTimeService;
+    protected CUMarshalService cuMarshalService;
+    protected FileStorageService fileStorageService;
+
+    @Override
+    public List<SupplierSyncMessage> getJaggaerContractsDto(JaggaerContractUploadProcessingMode processingMode,
+            Date processingDate, int maximumNumberOfSuppliersPerListItem) {
+        List<Supplier> suppliers = getAllVendorsToUploadToJaggaer(processingMode, processingDate);
+        LOG.info("getJaggaerContractsDto found {} suppliers.", suppliers.size());
+        return buildSupplierSyncMessageList(suppliers, maximumNumberOfSuppliersPerListItem);
+    }
+
+    private List<Supplier> getAllVendorsToUploadToJaggaer(JaggaerContractUploadProcessingMode processingMode,
+            Date processingDate) {
+        List<Supplier> suppliers = new ArrayList<>();
+
+        /*
+         * @todo fully implement this function in KFSPTS-28266
+         * 
+         * for now just create some testing data
+         */
+
+        suppliers.add(buildTestSupplier("13456-0", "Acme test company", "123 main street", "321 foo lane"));
+        suppliers.add(buildTestSupplier("13654-0", "foo test comapany", "789 main street", "456 foo lane"));
+        suppliers.add(buildTestSupplier("13456-1", "Acme test company part 2", "951 main street", "753 foo lane"));
+
+        return suppliers;
+    }
+
+    private Supplier buildTestSupplier(String erpNumber, String name, String addressLine1, String addressLine2) {
+        Supplier supplier = new Supplier();
+        supplier.setActive(JaggaerBuilder.buildActive(JaggaerConstants.YES));
+        supplier.setErpNumber(JaggaerBuilder.buildERPNumber(erpNumber));
+        supplier.setName(JaggaerBuilder.buildName(name));
+        supplier.setAddressList(new AddressList());
+        supplier.getAddressList().getAddress().add(buildTestAddress(addressLine1));
```

Comment:
Since there a number of variable naming issues like this I think this would be best handled in a follow up user story.  If you agree, I will create. a jira to address the naming conventions

Reply:
Deferring the DTO property name cleanup to a future user story is fine with me. Thanks!

Reply:
I created https://jira.cornell.edu/browse/KFSPTS-28468 to work on these changes.  You are a watcher on it.